### ISSUE TITLE
Getting stucked in calibration

### ISSUE THREAD

**Issue by satyagupta** [diggit/xlyrx#1]
Thanks for this fantastic code base. I am trying to rebuild a diy frsky receiver based on your frimware. Based on very little documentation and resources for CC2500 i really appreciate the efforts you have taken to write this firmware. 

What is happening right now, is the moment interrupt is attached on the GD0 pin while calibrating the mcu gets stucked in the callback. 

While checking on oscilloscope 2500 takes some time from going to idel to scal and during this the GD0 generates a square wave of very high freuqency, which is causing this. 

Any pointers how to get around this? Is there any way we can connect via email/discorrd/slack. I really need some guidance help on this to get it working. 

TIA

**Comment by diggit** [comment]
Hi,
I am not sure I'll be able to help. This project i 5 years old any all knowledge of CC2500 or FrSky (D?) protocol is no longer in my head. I never has issue with spammed IRQ pin as much as I remember. If it's not described in datasheet of CC2500, I'd guess it's something like missing pull-up or so.

You might be interested in project [DIY-Multiprotocol-TX-Module](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module).
It's still active and most probably working better.

**Comment by satyagupta** [comment]
Yea i checked that too but its too confusing and based on arduino or around it. I want it for a ARM mcu 

Thanks for responding on this :) i really appreciate your reply

**Comment by diggit** [comment]
Yeah, I am also not a big fan of Arduino based projects. Despite that, Arduino is basically C++, easy to manage libraries and one (too)simple code editor. It also supports ARM base MCUs. I am not telling you to switch to Arduino, just use it as guidance in your journey.
BTW, I switched to C++, CMake and VScode (cortex-debug) combo sometime ago.

**Comment by satyagupta** [comment]
I am planning to do the move :) 
Please share any resources if you have :)
